### PR TITLE
save filebacked modes

### PR DESCRIPTION
**Pull request by abuts**
This is part of the modifications for `save` for sqw files prepared to work with various versions of filebacked files. 

The main thing implemented here is `binfile_v4_common.put_new_blocks_values` method which allows saving various changed `sqw` object  components (any level 1-2 field of `sqw` object) defined in `faccess_sqw_v4` loader into existing `sqw` file. 

Started modifying `save` algorithm on the basis of these changes, but this job is not yet completed. 
As it should be working code, worth doing intermediate PR. 

As some `save` tests were written, I have also fixed multiple pathes in `save`. (All previously existing and couple of new). Just to fix existing tests, so `save` looks like almost completed too. 
But will be clearly finished after more tests are written to ensure everything there works as expected.

### CODE REVIEW

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/sqw/file_io/class_helpers/@pix_data_block/pix_data_block.m [hunk 1]
```diff
@@ -102,7 +104,7 @@
                 sqw_obj_to_set = obj.set_subobj(sqw_obj_to_set,subobj);
             end
         end
-        function obj =calc_obj_size(obj,sqw_obj,nocache)
+        function obj =calc_obj_size(obj,sqw_obj,nocache,varargin)
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/class_helpers/@pix_data_block/pix_data_block.m:107]: This is the disadvantage of revieving changes from web difference :). The method is `pix_data_block` overload of generic `calc_obj_size` method of parent `data_block` and this fact clearly stated in the row 108.  As this row have not changed, you have not noticed it. 

The parent `calc_obj_size` method have changed adding `check_size` input value. As this check does not make sence for `pix_data_block` (`pix.data` are not serialized) the overload ignores this property. But I still have to adhere to the calling interface so need to add `varargin` for code not to throw `too many arguments`

File: horace_core/sqw/file_io/class_helpers/@data_block/private/calc_obj_size_.m [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+% check_size
+%          -- if present and true, assumes that the size and
+%             position of the object was identified before
+%             (e.g. using nocache option) and are all known.
+%             Сhecks serialized size of the object against the
+%             size of the array calculated while serializing
+%             to ensure that precaclulatrions were performed
+%             correctly.
+%
+subobj = obj.get_subobj(sqw_obj);
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/class_helpers/@data_block/private/calc_obj_size_.m:22]: I am not serializing the whole sqw object here. I am pickin up sub-object only (part of `sqw` as `sqw.property.property` i.e. `sqw.experiment_data.instruments`). Method description modified.

File: horace_core/sqw/file_io/class_helpers/@blockAllocationTable/private/place_undocked_blocks_.m [hunk 3]
```diff
@@ -0,0 +1,56 @@
+function obj = place_undocked_blocks_(obj,obj_to_write,nocache)
+% replace contents of unlocked blocks with the contents of the
+% input object and found places of these blocks within the BAT.
+%
+% Should work after clear_unocked_blocks was called, as clear_unocked_blocks
+% calculates free spaces left after old blocks were
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/class_helpers/@blockAllocationTable/private/place_undocked_blocks_.m:6]: Difficult to answer something to this. Typo fixed but `clear_ulcocked_blocks` removes blocks with `lock == false` and `place_unDocked_blocks` find places for undocked blocks i.e. blocks with `position == 0` (undocked).

File: horace_core/sqw/file_io/class_helpers/@blockAllocationTable/private/merge_adjusent_space_.m [hunk 4]
```diff
@@ -0,0 +1,26 @@
+function fs_new = merge_adjusent_space_(fs)
+% merge together free space which occurs after block removal and
+% when blocks were located one after another. Get continuous chunks of
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/class_helpers/@blockAllocationTable/private/merge_adjusent_space_.m:3]: I am removing merging `contiguous` blocks and creating `continuous` space (without breaks) May be there is better word for this.

File: horace_core/sqw/file_io/class_helpers/@blockAllocationTable/blockAllocationTable.m [hunk 5]
```diff
@@ -281,6 +284,36 @@
             %       not been initialized
             pos = get_block_pos_(obj,block_name_or_class);
         end
+        function obj = place_undocked_blocks(obj,obj_to_write,nocache)
+            % replace contents of blocks which have not been already placed
+            % (position = 0) with the contents taken from the input object
+            % and found places of these blocks within the BAT.
+            %
+            % Should work after clear_unocked_blocks was called, as clear_unocked_blocks
```
[the remark this answers is not in the document]
Comment (abuts) [reply horace_core/sqw/file_io/class_helpers/@blockAllocationTable/blockAllocationTable.m:292]: clear unlocked, place undocked

**Review by cmarooney-stfc — APPROVED** [review]

**Pull request merged**